### PR TITLE
Insert variables in formio config

### PR DESCRIPTION
**Pull request by sergei-maertens**
Related to #1708 

**Changes**

* Added isolated django template backend instance for user-input rendering
* Added formio test utils/assertions to verify component properties
* Added service function to inject variables values into formio configuration (using the django template engine)
* Hooked up variable injection into the form logic evaluation function

### CODE REVIEW

File: src/openforms/formio/variables.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    :arg configuration: A dictionary containing the static Formio configuration (from
+      the form designer)
+    :arg values: A mapping of variable name to its value (Python native objects)
+    :returns: None - this function mutates the datastructures in place
+
+    .. todo:: Support getting non-string based configuration from variables, such as
+       `validate.required` etc.
+    """
+    for component in iter_components(configuration=configuration, recursive=True):
+        for property_name in SUPPOORTED_TEMPLATE_PROPERTIES:
```
Comment (SilviaAmAm) [inline src/openforms/formio/variables.py:71]:
```suggestion
        for property_name in SUPPORTED_TEMPLATE_PROPERTIES:
```

File: src/openforms/formio/variables.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+from openforms.template import render_from_string
+from openforms.typing import DataMapping, JSONObject, JSONValue
+
+from .utils import iter_components
+
+logger = logging.getLogger(__name__)
+
+
+SUPPOORTED_TEMPLATE_PROPERTIES = (
```
Comment (SilviaAmAm) [inline src/openforms/formio/variables.py:13]:
```suggestion
SUPPORTED_TEMPLATE_PROPERTIES = (
```

**Review by SilviaAmAm — APPROVED** [review]
Looks good!

I was wondering about rendering the updated configuration (for pdf report, confirmation email, summary page ... ). It should work already since the renderer calls the get_dynamic_configuration. Should we add a test that the rendering contains the correctly injected configuration?

**Comment by sergei-maertens** [comment]
> Looks good!
> 
> I was wondering about rendering the updated configuration (for pdf report, confirmation email, summary page ... ). It should work already since the renderer calls the get_dynamic_configuration. Should we add a test that the rendering contains the correctly injected configuration?

Good suggestion, I'll do that! I'm still looking in getting this PR's coverage up to 100% too :)

**Pull request merged**